processing/src/main/java/io/druid/segment/QueryableIndexStorageAdapter.java

```diff
@@ -327,7 +394,34 @@ public Cursor apply(final Long input)
                   {
                     private final Offset initOffset = offset.clone();
                     private final DateTime myBucket = gran.toDateTime(input);
-                    private Offset cursorOffset = offset;
+                    private final CursorOffsetHolder cursorOffsetHolder = new CursorOffsetHolder();
+                    {
+                      cursorOffsetHolder.set(offset);
```

Comment:
I think this is unnecessary, because you force a call to `reset()` on construction anyway.